src/logs/diagnostic-channel/initialization.ts

```diff
@@ -4,14 +4,17 @@
 // Don't reference modules from these directly. Use only for types.
 import * as DiagChannelPublishers from "diagnostic-channel-publishers";
 import { Logger } from "../../shared/logging";
+import { ShimJsonConfig } from "../../shim/jsonConfig";
 
 const TAG = "DiagnosticChannel";
 let isInitialized = false;
 
 export function enablePublishers() {
     // Only register monkey patchs once
-    if (!isInitialized) {
+    if (!isInitialized && (ShimJsonConfig.getInstance().noDiagnosticChannel !== true)) {
         isInitialized = true;
+        const individualOptOuts: string = ShimJsonConfig.getInstance().noPatchModules;
```

Comment:
Why are you adding the check here? you should be checking for config options in the subscriber code not here in publishers side

Reply:
Rational was that we used to disable diagnostic channels on the publisher side, however I agree it makes sense to disable them at a subscriber level. Will update that.

Reply:
Refactored this to use the publisher's `enable` functions (specifically not calling them if noDiagnosticChannel is set to true) in the same way as `noPatchModules`.

Reply:
Not seeing those changes here, is there a commit missing?

Reply:
Changed such that we still do this check on the publisher side (line 26 is where we do the check for noDiagnosticChannel). It takes advantage of the same `enable()` methods each of the subscribers has. If we don't want to do this check for noDiagnosticChannel here we could move the logic for both noDiagnosticChannel and noPatchModules to each individual subscriber, but after looking into moving the code to the subscribers it seems like it'd make this logic more complicated than necessary.